### DISCUSSION TITLE
Write out TableRelation with table field

### DISCUSSION THREAD

**Discussion by christianbraeunlich** [microsoft/alguidelines#192] [BC Best Practices]
To ensure that the reading flow of developers is not interrupted, it is recommended that the table field is also written out in the TableRelation property. This takes the guessing out and you don't have to jump into the table to figure out which is the first pk-field used by the TableRelation property.

### Bad code

```al
field(3007; "Related No."; Code[20])
{
    TableRelation = "Sales Line";
}
```

### Good code

```al
field(3007; "Related No."; Code[20])
{
    TableRelation = "Sales Line"."No.";
}
```

What do you think?

**Comment by waldo1001** [comment]
Very valid!

**Comment by KOTRET** [comment]
Somewhere, maybe from the good ol' days, I've read that the field should be omitted if it is the primary key.
Recently when writing some API pages, I've stumbled across a curiosity:
if the TableRelation contains the field, the api-fields referencing the other table could be expanded (without any extra code) automatically, whereas the API-field without the field defined in the relation could'nt, because BC wants to use the field-content in order to join the referenced entity, but tries to handle it a GUID (SystemId)!

**Comment by NKarolak** [comment]
Since programming in VS Code, I always add the PK field to the table relation, but for another reason: when I invoke "Show references" on the PK field, I want the table relations to be listed.
